contracts/Richman.sol

```diff
@@ -0,0 +1,167 @@
+        require(_addr == owner);
+        _;
+    }
+    
+    modifier notOwner(address _addr) {
+        require(_addr != owner);
+        _;
+    }
+    
+    /** @dev Checks if enough of free tokens to lent
+      * @param _amount Amount needed to check
+    */
+    modifier amountIsAvailable(uint _amount) {
+        require(freeTokens >= _amount);
+        _; 
+    }
+    
+    /** @dev Checks borrower has borrowed tokens
+      * @param _addr Address of borrower to check
+    */
+    modifier hasBorrowedTokens(address _addr) {
+      require(ledger[_addr] > 0);
+        _;
+    }
+    
+    modifier checkedForBorrowOverflow(address _addr, uint _amount) {
+        require( (freeTokens - _amount) < freeTokens &&
+                ledger[msg.sender] < (ledger[msg.sender] + _amount) );
+                
+        _;
+    }
+    
+     modifier checkedForPayBackUnderflow(address _addr, uint _amount) {
+        uint freeTokensWithPayBack = freeTokens + _amount;
+        uint borrowedAmount = ledger[msg.sender];
+
+        require( (freeTokensWithPayBack - freeTokens) == _amount &&
+                (borrowedAmount - _amount) < borrowedAmount );
+                
+        _;
+    }
+    
+    modifier moreThanZero(uint _amount) {
+        require(_amount > 0);
+        _;
+    }
+    
+    modifier notInBlacklist(address _addr) {
+        require(!blacklist[_addr]);
+        _;
+    }
+    
+    modifier isValidAddress(address _addr) {
+        require(_addr != 0x0);
+        _;
+    }
+    
+    //  Events
+    event LogBorrowSuccess(address _addr, uint _amount);
+    event LogPayBackSuccess(address _addr, uint _amount, string _str, uint _debt);
+    event LogTotalSupplyUpdateStatus(bool success);
+    event LogFallbackCalled(address _addr, string _str);
+    
+    
+    function Richman(uint _totalSupply) public payable {
+        owner = msg.sender;
+        
+        totalSupply = _totalSupply;
+        freeTokens = _totalSupply;
+    }
+    
+    function () payable public { 
+      var saluteStr = "Richman fallback was called by ";
+      if (msg.value > 0) {
+        saluteStr = "Richman fallback ( with ethers :P ) was called by ";
+      }
+
+      LogFallbackCalled(msg.sender, saluteStr);
+     }
+    
+    //  set new value
+    function updateTotalSupply(uint _amount) public isOwner(msg.sender) {
+        uint lentTokens = totalSupply - freeTokens;
+        
+        //  new total supply must be more or equal to already lent tokens
+        if (lentTokens > _amount) {
+            LogTotalSupplyUpdateStatus(false);
+            revert();
+        }
+        
+        totalSupply = _amount;
+        freeTokens = totalSupply - lentTokens;
+        LogTotalSupplyUpdateStatus(true);
+    }
+    
+    function updateAddressAsBlacklisted(address _addr, bool isBlacklisted) public
+        isOwner(msg.sender)
+        isValidAddress(_addr) {
+            if (isBlacklisted) {
+                blacklist[_addr] = true;
```

Comment:
Difference is - if not blacklisted, I `delete(blacklist[_addr]);`, so only blacklisted addresses are present in mapping. If I use `blacklist[_addr] = isBlacklisted`, then both types would be present.
Array would be more expensive, because of loops.

Reply:
`delete` is the exact same as `= false`